### ISSUE TITLE
 a long time to execute

### ISSUE THREAD

**Issue by cthed2** [CraftJarvis/RAT#4]
I tried the code you propose, but it takes a long time to execute (after 20 minutes I canceled it). I don't know the reason, maybe I am using different versions of the libraries.

**Comment by zhwang4ai** [comment]
Yes, due to the fact that RAT operates by causally retrieving and reasoning, it necessitates a longer runtime. This aspect is indeed a direction for future optimization on our part. Currently, the error primarily stems from webpage access timeouts. To address this, we've posted an updated demo version at https://huggingface.co/spaces/jeasinema/RAT, where we've integrated a run_with_timeout feature to ensure the process can continue in the event of such errors. We plan to update this location with the new version of the code moving forward.

**Comment by cthed2** [comment]
I understand. Can I use a text input query instead of web pages? Specifically, the transcription of a podcast.

Do you have any example to guide me?

The idea is to make it simple with an input text and maybe I can help you with timing optimizations or some suggestions in general.

Note: I use "whisper" for the transcription.

Greetings.

**Comment by cxb123d** [comment]
Have you reproduced the entire code
